v6.0.0 alpha.3 (#6495)
mui/mui-x: CHANGELOG.md

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## v6.0.0-alpha.3

_Oct 13, 2022_

We'd like to offer a big thanks to the 8 contributors who made this release possible. Here are some highlights ✨:

> - 🚀 Simplify `useGridApiContext` and `useGridApiRef` type overrides (#6423) @cherniavskii

Author: Let me know if we should highlight something else too?

Reviewer: I wouldn't highlight #6423, since it's an internal change and doesn't impact the users

> - 🚀 Simplify `useGridApiContext` and `useGridApiRef` type overrides (#6423) @cherniavskii

Reviewer 2: We clearly need a highlight for the new fields

@joserodolfofreitas @LukasTy @alexfauquette 

First draft:

- ⌚️ New components to edit dates and times with the beyboard, without using any modal or dropdown UI.
  Please check out our [documentation](https://next.mui.com/x/react-date-pickers/fields/) to discover those new components.

  - [`DateField`](https://next.mui.com/x/react-date-pickers/date-field/) to edit a single date
  - [`TimeField`](https://next.mui.com/x/react-date-pickers/time-field/)  to edit a single time
  - [`DateTimeField`](https://next.mui.com/x/react-date-pickers/date-time-field/)  to edit a single date and time
  - [`MultiInputDateRangeField` / `SingleInputDateRangeField`](https://next.mui.com/x/react-date-pickers/date-range-field/) to edit a date range with two inputs
  - [`MultiInputTimeRangeField`](https://next.mui.com/x/react-date-pickers/time-range-field/) to edit a time range with two inputs
  - [`MultiInputDateTimeRangeField`](https://next.mui.com/x/react-date-pickers/date-time-range-field/) to edit a date time range with two inputs
  
  ⚠️ These components are unstable.
  They might receive breaking changes on their props to have the best components possible by the time of the stable release.

Reviewer 3: Minor tweaking:

- ⌚️ New components to edit date and time with <kbd>keyboard</kbd>—without using any modal or dropdown UI.
  Please check out our [documentation](https://next.mui.com/x/react-date-pickers/fields/) to discover those new components.

  - [`DateField`](https://next.mui.com/x/react-date-pickers/date-field/) to edit date
  - [`TimeField`](https://next.mui.com/x/react-date-pickers/time-field/)  to edit time
  - [`DateTimeField`](https://next.mui.com/x/react-date-pickers/date-time-field/)  to edit date and time
  - [`MultiInputDateRangeField` / `SingleInputDateRangeField`](https://next.mui.com/x/react-date-pickers/date-range-field/) to edit date range
  - [`MultiInputTimeRangeField`](https://next.mui.com/x/react-date-pickers/time-range-field/) to edit time range with two inputs
  - [`MultiInputDateTimeRangeField`](https://next.mui.com/x/react-date-pickers/date-time-range-field/) to edit date and time range with two inputs
  
  ⚠️ These components are unstable.
  They might receive breaking changes on their props to have the best components possible by the time of the stable release.

Author: Added

- 📝 Allow to limit to one filter per column for `DataGridPro` and `DataGridPremium` (#6333) @MBilalShafi
- 🎁 Support pasting in pickers @flaviendelangle
- 📚 Documentation improvements
- 🐞 Bugfixes

### `@mui/x-data-grid@v6.0.0-alpha.3` / `@mui/x-data-grid-pro@v6.0.0-alpha.3` / `@mui/x-data-grid-premium@v6.0.0-alpha.3`

> #### Changes

Reviewer 2: To have the breaking change automatically copied here (and avoid mistakes), I do the following structure in your PR description

```
## Changelog

#### Breaking changes
```

> #### Changes

Author: Breaking changes added

  +  componentsProps={{ desktopTrapFocus: { isEnabled: () => false }}}
  />
  ```
- The view components allowing to pick a date or parts of a date without an input have been renamed to better fit their usage:

  ```diff
  -<CalendarPicker {...props} />
  +<DateCalendar  {...props} />
  ```

  ```diff
  -<DayPicker {...props} />
  +<DayCalendar  {...props} />
  ```

  ```diff
  -<CalendarPickerSkeleton {...props} />
  +<DayCalendarSkeleton  {...props} />
  ```

  ```diff
  -<MonthPicker {...props} />
  +<MonthCalendar  {...props} />
  ```

  ```diff
  -<YearPicker {...props} />
  +<YearCalendar  {...props} />
  ```

- Component names in the theme have changed as well:

  ```diff
  -MuiCalendarPicker: {
  +MuiDateCalendar: {
  ```

  ```diff
  -MuiDayPicker: {
  +MuiDayCalendar: {
  ```

  ```diff
  -MuiCalendarPickerSkeleton: {
  +MuiDayCalendarSkeleton: {
  ```

  ```diff
  -MuiMonthPicker: {
  +MuiMonthCalendar: {
  ```

  ```diff
  -MuiYearPicker: {
  +MuiYearCalendar: {
  ```

#### Changes

- [DatePicker] Allows to fix the number of week displayed (#6299) @alexfauquette
- [DateRangePicker] Fix calendar day outside of month layout shifting on hover (#6448) @alexfauquette

> - [fields] Prepare the field exports for the public release (#6467) @flaviendelangle

Reviewer 4: ```suggestion
- [fields] New components: `MultiInputDateTimeRangePicker` and `MultiInputTimeRangePicker` (#6392) @alexfauquette
- [fields] Prepare the field exports for the public release (#6467) @flaviendelangle
```

Author: Updated

- [pickers] Replace the `Picker` prefix in the view component by `Calendar` (eg: `MonthPicker` => `MonthCalendar`) (#6389) @flaviendelangle
- [pickers] Support pasting on fields (#6364) @flaviendelangle
- [pickers] Use slots in the mobile and desktop wrappers instead of `XXXComponent` and `XXXProps` (#6381) @flaviendelangle

### Docs

- [docs] Add migration to DataGrid v6 page (#6235) @m4theushw
- [docs] Create first publishable version of the field doc (#6323) @flaviendelangle
- [docs] Fix trailing spaces in the readme @oliviertassinari
- [docs] New page for the pickers: Validation (#6064) @flaviendelangle

### Core

- [core] Add CodeQL workflow (#6387) @DanailH
- [core] Add missing breaking change to the changelog (#6471) @flaviendelangle
- [core] Fix playground structure (#6466) @LukasTy
- [core] Fix tests for pasting on fields (#6465) @flaviendelangle
- [core] Remove absolute link (#6420) @flaviendelangle
- [core] Remove unused `react-text-mask` package (#6408) @LukasTy
- [core] Send explicit warning when dayjs locale is not found (#6424) @alexfauquette
- [core] Test validation on textfield and date views (#6265) @alexfauquette

> - New components: `MultiInputDateTimeRangePicker` and `MultiInputTimeRangePicker` (#6392) @alexfauquette

Reviewer 4: ```suggestion
```

Reviewer 4: Sorry, wrong PR title 🙈

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## v6.0.0-alpha.3

_Oct 13, 2022_

We'd like to offer a big thanks to the 8 contributors who made this release possible. Here are some highlights ✨:

- 📚 Documentation improvements

From https://github.com/mui/mui-x/pull/6381

> 

Reviewer 4: ```suggestion
```

#### Breaking changes

> - All the props used by the mobile and desktop wrappers to override components or components props has been replaced by component slots. You can find more information about this pattern in the [MUI Base documentation](https://mui.com/base/getting-started/usage/#shared-props).

Reviewer 4: ```suggestion
- All the props used by the mobile and desktop wrappers to override components or components' props have been replaced by component slots. You can find more information about this pattern in the [MUI Base documentation](https://mui.com/base/getting-started/usage/#shared-props).
```

  Some of the names have also been prefixed by `desktop` when it was unclear that the behavior was only applied on the desktop version of the pickers (or the responsive version when used on a desktop).

>   The `DialogProps` prop has been replaced by a `dialog` component props slot on a responsive an mobile pickers:

Reviewer 4: ```suggestion
  The `DialogProps` prop has been replaced by a `dialog` component props slot on responsive and mobile pickers:
```

  ```diff
  // Same on DesktopDatePicker, DateTimePicker, DesktopDateTimePicker, 
  // TimePicker, DesktopTimePicker, DateRangePicker and DesktopDateRangePicker.
  <DatePicker
  -  PopperProps={{ onClick: handleClick }}
  +  componentsProps={{ popper: { onClick: handleClick }}}
  />
  ```

  The `TransitionComponent` prop has been replaced by a `DesktopTransition` component slot on all responsive and desktop pickers:

  ```diff
  // Same on DesktopDatePicker, DateTimePicker, DesktopDateTimePicker, 
  // TimePicker, DesktopTimePicker, DateRangePicker and DesktopDateRangePicker.
  <DatePicker
  -  TransitionComponent={Fade}
  +  components={{ DesktopTransition: Fade }}
  />
  ```

  The `TrapFocusProps` prop has been replaced by a `desktopTrapFocus` component props slot on all responsive and desktop pickers:

  ```diff
  // Same on DesktopDatePicker, DateTimePicker, DesktopDateTimePicker, 
  // TimePicker, DesktopTimePicker, DateRangePicker and DesktopDateRangePicker.
  <DatePicker
  -  TrapFocusProps={{ isEnabled: () => false }}
  +  componentsProps={{ desktopTrapFocus: { isEnabled: () => false }}}
  />
  ```
 
From https://github.com/mui/mui-x/pull/6389

> 

Reviewer 4: ```suggestion
```

  Please check out our [documentation](https://next.mui.com/x/react-date-pickers/fields/) to discover those new components.
  - [`DateField`](https://next.mui.com/x/react-date-pickers/date-field/) to edit date
  - [`TimeField`](https://next.mui.com/x/react-date-pickers/time-field/)  to edit time
  - [`DateTimeField`](https://next.mui.com/x/react-date-pickers/date-time-field/)  to edit date and time
  - [`MultiInputDateRangeField` / `SingleInputDateRangeField`](https://next.mui.com/x/react-date-pickers/date-range-field/) to edit date range
  - [`MultiInputTimeRangeField`](https://next.mui.com/x/react-date-pickers/time-range-field/) to edit time range with two inputs
  - [`MultiInputDateTimeRangeField`](https://next.mui.com/x/react-date-pickers/date-time-range-field/) to edit date and time range with two inputs
  
  ⚠️ These components are unstable.
  They might receive breaking changes on their props to have the best components possible by the time of the stable release.

- 📝 Allow to limit to one filter per column for `DataGridPro` and `DataGridPremium` (#6333) @MBilalShafi
- 🎁 Support pasting in pickers @flaviendelangle

> - 📚 Documentation improvements

Reviewer 2: Should we add a line for the new validation page ? 
We added one for the `apiRef` page a few months back

```suggestion
- 📚 New [page describing the validation props on each picker](https://next.mui.com/x/react-date-pickers/validation/) (#6064) @flaviendelangle

- 📚 Documentation improvements
```

- [DateRangePicker] Fix calendar day outside of month layout shifting on hover (#6448) @alexfauquette
- [fields] New components: `MultiInputDateTimeRangePicker` and `MultiInputTimeRangePicker` (#6392) @alexfauquette
- [fields] Prepare the field exports for the public release (#6467) @flaviendelangle
- [fields] Support paste in single section (#6422) @alexfauquette
- [pickers] Add field placeholders to the locale (#6337) @flaviendelangle
- [pickers] Do not use `Partial` for `components` and `componentsProps` props (#6463) @flaviendelangle
- [pickers] New component: `DateRangeCalendar` (#6416) @flaviendelangle
- [pickers] Replace the `Picker` prefix in the view component by `Calendar` (eg: `MonthPicker` => `MonthCalendar`) (#6389) @flaviendelangle
- [pickers] Support pasting on fields (#6364) @flaviendelangle
- [pickers] Use slots in the mobile and desktop wrappers instead of `XXXComponent` and `XXXProps` (#6381) @flaviendelangle

### Docs

- [docs] Add migration to DataGrid v6 page (#6235) @m4theushw
- [docs] Create first publishable version of the field doc (#6323) @flaviendelangle
- [docs] Fix trailing spaces in the readme @oliviertassinari

> - [docs] New page for the pickers: Validation (#6064) @flaviendelangle

Reviewer 2: ```suggestion
- [docs] New page for the pickers: Validation (#6064) @flaviendelangle
- [docs] Organize migration pages (#6480) @flaviendelangle
```
